hack/validate-migration.sh

```diff
@@ -249,25 +249,35 @@ is_migration_file() {
 # discovered, it will be checked later. # No argument is required. Function
 # inspects changed files from current branch directly.
 list_migration_files() {
-    git diff --name-status "$(git merge-base HEAD "$DEFAULT_BRANCH")" | \
-        while read -r status origin_path; do
-            if ! is_migration_file "$origin_path"; then
-                continue
-            fi
-            case "$status" in
-                A)  # file is added
+    local task_dir_path  # Used as key for TASK_MIGRATIONS
+    local file_list
+    file_list=$(git diff --name-status "$(git merge-base HEAD "$DEFAULT_BRANCH")")
+    local seen=
+    while read -r status origin_path; do
+        if ! is_migration_file "$origin_path"; then
+            continue
+        fi
+        case "$status" in
+            A)  # file is added
+                task_dir_path=$(awk -F '/' '{ OFS = "/"; print $1, $2 }' <<<"$origin_path")
+                if grep -q "$task_dir_path" <<<"$seen"; then
```

Comment:
I think this will cause problems

```bash
$ grep -q "task/buildah" <<< "task/buildah-remote task/some-other-task" && echo already seen                                                       
already seen
```

Maybe change the `seen` variable to be multiline and check like this?

```bash
grep -q "^${task_dir_path}$" <<< "$seen"
```

Reply:
Good catch. Updated.